Intake Assist v1

simonstoryparker opened the pull request.

simonstoryparker committed:
Add translation based intake assist

simonstoryparker committed:
Add translation based intake assist

simonstoryparker committed:
Add translation based intake assist

simonstoryparker committed:
Add translation based intake assist

jonahsnider committed:
Zone based collision avoidance (#134)

* start new collision avoidance logic

* collision avoidance to be tested

* remove bandaid state collision avoidance

* change practice config to be 0.1 acceleration limit and velocity limit for testing

* remove bad box cause its dumb

* clamp current position to a zone if it is outside of all the zones

* change to testing config

* spotless

* logging and fix a test

* logging changes

* remove sticking straight out zone, and logging

* remove collision avoidance utils cause i dont use it

* fix code related to the removal of the straight out zone

* fix test, change zone, revert the velocity limit to be normal, make the collision update always

* spotless apply

---------

Co-authored-by: ryanknj5 <140574269+ryanknj5@users.noreply.github.com>

jonahsnider committed:
Fix unitialized fields in LightsSubsystem

jonahsnider committed:
Use lights.setDisabledState()

jonahsnider committed:
Log current position in collision avoidance

jonahsnider committed:
Adjust L1 scoring logic & lights while disabled

fcuellar13 committed:
coral l4 lineup higher by 2 in

jonahsnider committed:
Improved coral station intake

jonahsnider committed:
Warn when battery voltage below 12.5V

jonahsnider committed:
Tune collision avoidance boxes 1 & 2

jonahsnider commented on src/main/java/frc/robot/robot_manager/RobotManager.java:
```diff
@@ -327,6 +329,7 @@ protected void afterTransition(RobotState newState) {
         moveSuperstructure(ElevatorState.GROUND_CORAL_INTAKE, WristState.GROUND_CORAL_INTAKE);
         swerve.setSnapsEnabled(false);
```
delete this line, it is made redundant with the coral assist call

jonahsnider commented on src/main/java/frc/robot/swerve/SwerveSubsystem.java:
```diff
@@ -116,6 +117,10 @@ public void setFieldRelativeAutoSpeeds(ChassisSpeeds speeds) {
     sendSwerveRequest();
   }
 
+  public void setAssistSpeedsOffset(ChassisSpeeds speedsOffset) {
```
rename this function to make it clear that it accepts a robot relative speeds.


then, update the function to convert it to field relative and store that in the field. this is bc teleop speeds are field relative, so we should standardize around that

simonstoryparker replied:
intake assist speeds from `IntakeAssistUtil` are field relative already, we could make the name for whatever the util function is more clear

jonahsnider commented on src/main/java/frc/robot/swerve/SwerveSubsystem.java:
```diff
@@ -216,6 +221,16 @@ private void sendSwerveRequest() {
+                    teleopSpeeds.vxMetersPerSecond + assistSpeedsOffset.vxMetersPerSecond)
+                .withVelocityY(
+                    teleopSpeeds.vyMetersPerSecond + assistSpeedsOffset.vyMetersPerSecond)
+                .withRotationalRate(teleopSpeeds.omegaRadiansPerSecond)
```
for posterity we should also add the assist omega here, even if we don't plan on doing anything with it for now

jonahsnider commented on src/main/java/frc/robot/vision/game_piece_detection/GamePieceDetectionUtil.java:
```diff
@@ -104,4 +72,26 @@ public static double getRobotRelativeAngleToGamePiece(GamePieceResult visionResu
+    var robotRelativeGamePiecePose = calculateRobotRelativeTranslationFromCamera(visionResult);
+    var adjustedTranslationForIntake =
+        new Translation2d(
+            robotRelativeGamePiecePose.getX() - 0.762, robotRelativeGamePiecePose.getY());
```
this number should be a constant with a comment explaining it

jonahsnider commented on src/main/java/frc/robot/vision/game_piece_detection/GamePieceDetectionUtil.java:
```diff
@@ -104,4 +72,26 @@ public static double getRobotRelativeAngleToGamePiece(GamePieceResult visionResu
+  public static Translation2d calculateRobotRelativePoseToIntake(GamePieceResult visionResult) {
+    var robotRelativeGamePiecePose = calculateRobotRelativeTranslationFromCamera(visionResult);
+    return new Translation2d(
+        robotRelativeGamePiecePose.getX() - 0.762, robotRelativeGamePiecePose.getY());
```
same note about magic number

jonahsnider committed:
Add autofix.ci

renovate[bot] committed:
Update dependency gradle to v8.12.1 (#126)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency org.junit.jupiter:junit-jupiter to v5.11.4 (#61)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update plugin com.diffplug.spotless to v7.0.2 (#99)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

jonahsnider committed:
Magnetism rework (#136)

* move to branch

* tuning

* tuning

* FORMAT

* Adding tests + cleaning up MagnetismUtil.

1. Refactoring function for magnetism util to also take in a goal pose.
2. Adding tests to the magnetism util to test properties of the output
that we care about.

* magnetism tuning

* scoring pose offset and magnetism tuning

* [autofix.ci] apply automated fixes

* Add reef magnetism toggle

---------

Co-authored-by: Owen <117956892+Owen756@users.noreply.github.com>
Co-authored-by: Saikiran Ramanan <saikiranra@gmail.com>
Co-authored-by: Jordan <117786700+jordanjcoderman@users.noreply.github.com>
Co-authored-by: autofix-ci[bot] <114827586+autofix-ci[bot]@users.noreply.github.com>

jonahsnider committed:
Magnetism rework (#136)

* move to branch

* tuning

* tuning

* FORMAT

* Adding tests + cleaning up MagnetismUtil.

1. Refactoring function for magnetism util to also take in a goal pose.
2. Adding tests to the magnetism util to test properties of the output
that we care about.

* magnetism tuning

* scoring pose offset and magnetism tuning

* [autofix.ci] apply automated fixes

* Add reef magnetism toggle

---------

Co-authored-by: Owen <117956892+Owen756@users.noreply.github.com>
Co-authored-by: Saikiran Ramanan <saikiranra@gmail.com>
Co-authored-by: Jordan <117786700+jordanjcoderman@users.noreply.github.com>
Co-authored-by: autofix-ci[bot] <114827586+autofix-ci[bot]@users.noreply.github.com>

jonahsnider committed:
Delete failing collision avoidance tests

jonahsnider committed:
Use WPILib builtins for collision avoidance (#140)

* Use WPILib builtins for collision avoidance

* Log coordinates of collision boxes

jordanjcoderman committed:
fix teleopspeeds 0 when magnetism disabled

ryanknj5 committed:
tune collision boxes

simonstoryparker committed:
Fix reef pipe offset

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

fcuellar13 committed:
Adjust reef pipe red cords

simonstoryparker committed:
Fix reef pipe rotations

simonstoryparker committed:
Format

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

ryanknj5 committed:
tune collision box for station zone

fcuellar13 committed:
Adjust angles for pipe coords

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

simonstoryparker committed:
Fix merge issues

jordanjcoderman committed:
lineup to offset pose

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

simonstoryparker committed:
Upload default fmap file

simonstoryparker committed:
Add force forward feature to intake assist

simonstoryparker committed:
Add force forward feature to intake assist

fcuellar13 committed:
Add preload command for autos, update coords in 4 Piece auto & add waitForState conditon to AutoCommands

fcuellar13 committed:
Pathplanner coord changes

jonahsnider committed:
Fix name of FollowSegmentIndefinitely command

fcuellar13 committed:
remove useless timeout for auto command

fcuellar13 committed:
Update naming of auto to include last piece

jonahsnider committed:
Add names to commands

jonahsnider committed:
Improve command logging in BaseAuto

fcuellar13 committed:
Change structure of all l4coralLineup event markers

fcuellar13 committed:
add condtion to rehomeRoll command

fcuellar13 committed:
format

ryanknj5 committed:
collision avoidance tuning and change L1 wrist angle

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

simonstoryparker committed:
Make stowed position 5 degrees more forward

ryanknj5 committed:
edit collision avoidance tests

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

ryanknj5 committed:
change collision avoidance safe point to 40 degrees

simonstoryparker committed:
Make stowed position 50 degrees

simonstoryparker committed:
Change stowed angle to be 40

fcuellar13 committed:
Update naming of scoring requests to be accurate

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

jonahsnider committed:
Extract starting pose into BaseAuto

jonahsnider committed:
Use preallocated Rotation2d and Pose2d instances

simonstoryparker committed:
Tune intake assist

fcuellar13 committed:
add stow command to l4scoreAndReleaseCommand

fcuellar13 committed:
adjust scoring coord for first piece & raise elevator later in path for auto

fcuellar13 committed:
pathplanner coord changes

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

jonahsnider committed:
Comment out failing test

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

jonahsnider committed:
Smart stow (#148)

* Start smart stow after station intaking

* Fix smart stow functionality

* [autofix.ci] apply automated fixes

* Continue work on smart stow and add linear filter for rehoming

* Continue work on finishing smart stow and fixing rehome

* [autofix.ci] apply automated fixes

---------

Co-authored-by: autofix-ci[bot] <114827586+autofix-ci[bot]@users.noreply.github.com>
Co-authored-by: Jordan <117786700+jordanjcoderman@users.noreply.github.com>

simonstoryparker committed:
Make intake assist constants to improve clarity

simonstoryparker committed:
Remove redundancy in setting swerve snaps to false in same state intake assist enabled

simonstoryparker committed:
Add intake assist omega speeds for posterity

simonstoryparker committed:
Format

simonstoryparker committed:
Add hold trigger for intake assist, single press for regular intaking

simonstoryparker committed:
Format

fcuellar13 committed:
move JKL points

Owen756 committed:
Side station coral (#141)

* initial two side intake station

* format

* front station intake work

* format

* wrist merge change

* [autofix.ci] apply automated fixes

* format

---------

Co-authored-by: autofix-ci[bot] <114827586+autofix-ci[bot]@users.noreply.github.com>

jordanjcoderman committed:
fix red side pipe poses being incorrect

fcuellar13 committed:
fix logic for getting coral score direction for roll

jonahsnider committed:
Format with Spotless

Owen756 committed:
add intake station approach

simonstoryparker committed:
Format
